tools/importer.py

```diff
@@ -445,6 +448,123 @@ def execute(self, context):
+            return {'FINISHED'}
+
+        #try:
+        filename, extension = os.path.splitext(self.filepath)
+
+        if(extension == ".vmd"):
+            
+            #A dictionary to change the current model to MMD importer compatable temporarily
+            bonedict = {
+                "Chest":"UpperBody",
+                "Neck":"Neck",
+                "Head":"Head",
+                "Hips":"Center",
+                "Spine":"LowerBody",
+                
+                "Right wrist":"Wrist_R",
+                "Right elbow":"Elbow_R",
+                "Right arm":"Arm_R",
+                "Right shoulder":"Shoulder_R",
+                "Right leg":"Leg_R",
+                "Right knee":"Knee_R",
+                "Right ankle":"Ankle_R",
+                "Right toe":"Toe_R",
+                
+                
+                "Left wrist":"Wrist_L",
+                "Left elbow":"Elbow_L",
+                "Left arm":"Arm_L",
+                "Left shoulder":"Shoulder_L",
+                "Left leg":"Leg_L",
+                "Left knee":"Knee_L",
+                "Left ankle":"Ankle_L",
+                "Left toe":"Toe_L"
+                
+            }
+            
+            armature = Common.set_default_stage()
+            new_armature = armature.copy()
+            bpy.context.collection.objects.link(new_armature)
+            new_armature.data = armature.data.copy()
+            new_armature.name = "Cats MMD Rig Proxy"
+            new_armature.animation_data_clear()
+            
+            Common.unselect_all()
+            Common.set_active(new_armature)
+            Common.switch('OBJECT')
+            
+            for bone in new_armature.data.bones:
+                if bone.name in bonedict:
+                    bone.name = bonedict[bone.name]
+            
+            bpy.ops.mmd_tools.import_vmd(filepath=self.filepath,bone_mapper='RENAMED_BONES',use_underscore=True, dictionary='INTERNAL')
+            
+            #create animation for original if there isn't one.
+            if armature.animation_data == None :
+                armature.animation_data_create()
+            if armature.animation_data.action == None:
+                armature.animation_data.action = bpy.data.actions.new("MMD Animation")
+                
+                
+            #create animation for new if there isn't one.
+            if new_armature.animation_data == None :
+                new_armature.animation_data_create()
+            if new_armature.animation_data.action == None:
+                new_armature.animation_data.action = bpy.data.actions.new("EMPTY_SOURCE")
+            
+            active_obj = new_armature
+            ad = armature.animation_data
+            
+            #iterate through bones and translate them back, therefore blender API will change the animation to be correct.
+            for bone in new_armature.data.bones:
+                if bone.name in bonedict.values():
```

Comment:
Please create a set from bonedict.values() outside the for loop and check if bone.name is in that set. You should avoid repeatedly checking if elements are `in` list-like objects because in the worst-case, you have to check through every single element in the list every time you try to find if an element is in the list.